### ISSUE TITLE
Compilation errors

### ISSUE THREAD

**Issue by LostKin** [Mysteryem/Up_And_Down_And_All_Around#121]
I am trying to read your sources, to understand how did you make this. But when i am trying to compile the exact thing that i downloaded from github, i get this error:
`src/main/java/uk/co/mysterymayhem/gravitymod/client/renderers/RenderGravityEntityItem.java:67: error: transformModelCount(EntityItem,double,double,double,float,IBakedModel) has private access in RenderEntityItem
            return super.transformModelCount(itemIn, xPos, yPos, zPos, partialTicks, bakedModel);`
I checked, and yes, the method `RenderEntityItem.transformationModelCount` is private, while you trying to override it.

**Comment by Mysteryem** [comment]
Your development workspace doesn't seem to have been set up correctly, there is an access transformer that sets this method to public: https://github.com/Mysteryem/Up_And_Down_And_All_Around/blob/master/src/main/resources/META-INF/mysttmtgravitymod_at.cfg
As defined in the build.gradle: https://github.com/Mysteryem/Up_And_Down_And_All_Around/blob/f617e6969aefee13c8801eff274b4ef417e702bb/build.gradle#L98

**Comment by LostKin** [comment]
That helped, thanks, but i am still running in some issues(i think their on my side, not yours). But is you don't mind helping a newbie, my discord is "Sasha Drozdov#8680"

**Comment by Mysteryem** [comment]
tl;dr It would probably be easier to start from scratch than modify the existing code to support any Vec3 gravity direction.

A lot of the code is designed around only the cardinal directions because shortcuts can be taken to go between them. For example, it is a lot simpler to take normal player motion and adjust them to match the player's gravity. All you have to do is swap around the x, y and z and possibly negate them, e.g. (x,y,z) ->(y,-x,-z).

For upwards gravity, the simplest example:
Hold out your left hand with your palm facing upwards, close your hand and then extend your index finger forwards and your thumb to the left. In this example, your finger signifies "forwards", your thumb signifies "left" and your palm signifies "up".
Now, keep your finger pointing forwards and rotate your hand 180 degrees.
Your thumb should now be pointing to the right, but you know that it signifies "left", therefore, if you want to move the player when they're upside-down, you need to swap left and right.
Similarly, your palm is now facing down so you need to swap up and down too.

Of course, this isn't the only way to swap up and down, with your palm facing downwards, your index finger can point in any direction in 360 degrees. Another easy one you can do with your hand is to start with the same combination of index finger forwards and thumb to the left, then, keeping the thumb to your left, rotate your hand towards yourself so that your index finger is pointing towards you. Now left and right are still the same, but forwards and backwards are swapped along with up and down being swapped.

If you don't keep your thumb and index finger to 90 degree increments, then you start having to use trigonometry to figure out how movement will be affected.

Similar shortcuts can be made for adjusting player pitch and yaw in most cases, though I found that in one or two of the gravity directions no shortcut was possible and I had to use trigonometry.

The main player movement/collision code is almost entirely vanilla and that code works around expanding player bounding boxes and checking for intersections with bounding boxes of the world, reducing the player's intended movement until no intersection occurs. This vanilla code is limited to the 6 cardinal directions in which bounding boxes can expand, my code alters which direction gets expanded depending on the gravity direction.

There is an airships mod, Valkyrien Skies, that allows free rotation of the airship, in order to account for the blocks of the airship being rotated at any angle, the mod authors wrote their own collision code. I think their code may be the best place to start from if intending to make a gravity mod which allows for any Vec3 direction.
